NGitLab.Mock/Clients/MembersClient.cs

```diff
@@ -14,78 +17,118 @@ public MembersClient(ClientContext context)
     {
     }
 
+    public IEnumerable<Membership> OfProject(string projectId)
+    {
+        return OfProject(projectId, includeInheritedMembers: false);
+    }
+
+    public IEnumerable<Membership> OfProject(string projectId, bool includeInheritedMembers)
+    {
+        using (Context.BeginOperationScope())
+        {
+            var project = GetProject(projectId, ProjectPermission.View);
+            var members = project.GetEffectivePermissions(includeInheritedMembers).Permissions;
+            return members.Select(member => member.ToMembershipClient());
+        }
+    }
+
+    public GitLabCollectionResponse<Membership> OfProjectAsync(ProjectId projectId, bool includeInheritedMembers = false)
+    {
+        return GitLabCollectionResponse.Create(OfProject(projectId.ValueAsString(), includeInheritedMembers));
+    }
+
     public Membership AddMemberToProject(string projectId, ProjectMemberCreate projectMemberCreate)
     {
         using (Context.BeginOperationScope())
         {
             var project = GetProject(projectId, ProjectPermission.Edit);
             var user = Server.Users.GetById(projectMemberCreate.UserId);
 
-            CheckUserPermissionOfProject(projectMemberCreate.AccessLevel, user, project);
+            if (project.Permissions.Any(p => p.User.Id == user.Id))
+            {
+                throw new GitLabException("Member already exists")
```

Comment:
I don't know if this is considered a breaking change or not. I would call it a bug fix since it was not accurately emulating GitLab's behavior.

These changes didn't break any of the Mock tests, surprisingly. I guess the mock tests just avoid illegal behavior.

Reply:
I don't think that's too much of a concern. I'm ok with that.